wasmtime-wasi/src/host_impls.rs

```diff
@@ -1,5 +1,64 @@
-use super::wasm32;
+use super::host;
 
-pub fn wasmtime_ssp_proc_exit(rval: wasm32::__wasi_exitcode_t) {
+pub unsafe fn wasmtime_ssp_args_get(
+    argv_environ: &mut host::argv_environ_values,
+    argv: &mut [*mut host::char],
+    argv_buf: *mut host::char,
+) -> host::__wasi_errno_t {
+    for i in 0..(*argv_environ).argc {
+        let buf_off = isize::checked_sub(
+            *(*argv_environ).argv.offset(i as isize) as _,
+            (*argv_environ).argv_buf as _,
+        )
+        .expect("argv[i] - argv_buf overflows");
+        argv[i] = argv_buf.offset(buf_off);
+    }
+    argv[(*argv_environ).argc] = std::ptr::null_mut();
+    argv_buf.copy_from((*argv_environ).argv_buf, (*argv_environ).argv_buf_size);
+    host::__WASI_ESUCCESS as host::__wasi_errno_t
```

Comment:
The compiler complains here the return value is u32 but `host::__wasm_errno_t` is u16. I tried wrapping the `#define` declaration with the type via `((__wasm_errno_t)(0))` to force the u16 width in `host`, but this causes bindgen to drop the define entirely (https://github.com/rust-lang/rust-bindgen/issues/316).

Reply:
Yep, I stumbled upon this in #127 as well. In order to circumvent the issue, I've defined `host::__wasi_errno_t` errors manually in `host.rs` and blacklisted all the symbols in bindgen to avoid redefinitions. I reckoned that, since we're after moving everything over from C anyway, sooner or later we'll need to port the errors over as well, so why not do it now.

Anyhow, @heyitsanthony and @sunfishcode if you guys agree with the approach, @heyitsanthony you could wait for #127 to get merged and then you should be all set here. Of course, if you see better solutions to this (and I'm sure there are!), please do let me know, and I'll be happy to adjust!

Reply:
If I understand what's happening here, this is due to a limitation of bindgen and the wasmtime_ssp.h header. The approach in #127 fixes this by declaring the host errno values as type host::__wasi_errno_t, which looks like a good approach.